compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -509,15 +521,25 @@ getMatmulOrIGEMMLoweringConfigAndWorkgroupSize(
   attrs.emplace_back(StringAttr::get(context, "mma_kind"), kind);
   if (mustBeAligned) {
     Attribute useGlobalDma = IREE::GPU::UseGlobalLoadDMAAttr::get(context);
-    Attribute promotionArray[] = {useGlobalDma, useGlobalDma};
+    SmallVector<Attribute> promotionArray = {useGlobalDma, useGlobalDma};
+    SmallVector<int64_t> promotionList = {0, 1};
+    if (scaled) {
+      promotionArray.append({useGlobalDma, useGlobalDma});
+      promotionList.append({2, 3});
```

Comment:
This seems wrong that it is 2,3 if you are not promoting the result and 3,4 if you are not promoting the result on L540, anyway I am removing c promotion in a different PR and I will leave the scales as 3 and 4 in that one.